streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -60,8 +62,10 @@
 
     private boolean commitRequested = false;
     private boolean commitOffsetNeeded = false;
-
     private boolean requiresPoll = true;
+    private final Time time;
+    private final StreamsMetrics metrics;
+    private Sensor nodeCommitTimeSensor;
```

Comment:
By layer I mean three layers: thread, task, node, and for task I think just one sensor "commit" is good enough, for punctuating the node-level metrics should be sufficient to cover.